src/wp-includes/global-styles-and-settings.php

```diff
@@ -255,3 +258,14 @@ function ( $item ) {
 		}
 	}
 }
+
+/**
+ * Private function to clean the caches under the theme_json group.
+ *
+ * @since 6.1.2
+ * @access private
+ */
+function _wp_clean_theme_json_caches() {
```

Comment:
See my feedback in https://github.com/WordPress/wordpress-develop/pull/3556#discussion_r1071514752, this shouldn't be marked private.
```suggestion
 */
function wp_clean_theme_json_caches() {
```

Reply:
Until we finalize this api, it being private might be a good thing. We can always make this public later.

Reply:
@spacedmonkey We can't change function names in core. Even if we want to start it as private, we should still avoid the underscore prefix, that's discouraged in core anyway.

I don't see a good reason to mark this as private - it's not private anyway, and we can easily enhance what this function cleans without impacting how it's used. Pretty much all other core functions for cache cleanups are public as well.

Reply:
I know. But we can wrap in a public facing function. 
```
function wp_clean_theme_json_caches() {
   return _wp_clean_theme_json_caches();
}
```
This is done elsewhere in core. See 
```
	function wp_get_current_user() {
		return _wp_get_current_user();
	}
```

We can make it public later but not the other way around.

Reply:
Yes, but why even use an underscore? That is not a good pattern. Using `@access private` is the appropriate way to mark something as private.

Core has some underscore-prefixed functions, but most of them are old. In my years contributing to core several more seasoned committers have advised to not introduce underscore prefixed functions. We can always remove `@access private` later, but we can't change the name. Your workaround works, but why introduce a new function just to make something public, that isn't a good approach.

Reply:
> Core has some underscore-prefixed functions, but most of them are old.

- _filter_block_template_part_area - 5.9.0
- _get_block_templates_paths - 5.9.0
- _get_block_template_file - 5.9.0
- _flatten_blocks - 5.9.0
- _build_block_template_result_from_file - 5.9.0
- _wp_build_title_and_description_for_single_post_type_block_template - 6.1.0
- _wp_get_iframed_editor_assets - 6.0.0
- _resolve_home_block_template - 6.0.0

There were more examples of this, but you get the point. Underscore prefix is still actively used.

Reply:
Right, that doesn't mean it's a good practice. I did say _most_ of them are old :)

Per my explanations above, I don't think we should be using underscore prefixed functions anywhere. That's my take, based on what I learned from other committers with a lot more experience than myself.

Reply:
This is now already in core, so this comment thread is obsolete once `trunk` has been merged back into this PR.